gradio/chat_interface.py

```diff
@@ -77,6 +79,7 @@ def __init__(
             css: custom css or path to custom css file to use with interface.
             analytics_enabled: Whether to allow basic telemetry. If None, will use GRADIO_ANALYTICS_ENABLED environment variable if defined, or default to True.
             submit_btn: Text to display on the submit button. If None, no button will be displayed. If a Button object, that button will be used.
+            stop_btn: Text to display on the stop button, which replaces the submit_btn when the submit_btn or retry_btn is clicked and response is streaming. Clicking on the stop_btn will halt the chatbot response. If set to None, stop button functionality does not appear in the chatbot. If a Button object, that button will be used as the stop button (the Button should set visible=False).
```

Comment:
I think `submit_btn=None` is broken now

```bash
Traceback (most recent call last):
  File "/Users/freddy/sources/scratch/chatinterface_test.py", line 52, in <module>
    gr.ChatInterface(predict, submit_btn=None).queue().launch()
  File "/Users/freddy/sources/gradio/gradio/chat_interface.py", line 199, in __init__
    self._setup_events()
  File "/Users/freddy/sources/gradio/gradio/chat_interface.py", line 227, in _setup_events
    self._setup_stop_events(self.textbox.submit, submit_event)
  File "/Users/freddy/sources/gradio/gradio/chat_interface.py", line 330, in _setup_stop_events
    event_to_cancel.then(
  File "/Users/freddy/sources/gradio/gradio/events.py", line 139, in __call__
    dep, dep_index = self.trigger.set_event_trigger(
  File "/Users/freddy/sources/gradio/gradio/blocks.py", line 282, in set_event_trigger
    "outputs": [block._id for block in outputs],
  File "/Users/freddy/sources/gradio/gradio/blocks.py", line 282, in <listcomp>
    "outputs": [block._id for block in outputs],
AttributeError: 'NoneType' object has no attribute '_id'
```

With this


```python
gr.ChatInterface(predict, submit_btn=None).queue().launch()
```

Reply:
Fixed